src/ring_theory/ideal/operations.lean

```diff
@@ -443,6 +443,31 @@ lemma span_singleton_mul_le_span_singleton_mul {x y : R} {I J : ideal R} :
   span {x} * I ≤ span {y} * J ↔ ∀ zI ∈ I, ∃ zJ ∈ J, x * zI = y * zJ :=
 by simp only [span_singleton_mul_le_iff, mem_span_singleton_mul, eq_comm]
 
+lemma span_singleton_mul_right_mono [is_domain R] {x : R} (hx : x ≠ 0) :
+  span {x} * I ≤ span {x} * J ↔ I ≤ J :=
```

Comment:
One advantage of using [submodule.pointwise_distrib_mul_action](https://leanprover-community.github.io/mathlib_docs/algebra/module/submodule/pointwise.html#submodule.pointwise_distrib_mul_action) is that it doesn't require the semiring to be commutative, while [ideal.has_mul](https://leanprover-community.github.io/mathlib_docs/ring_theory/ideal/operations.html#ideal.has_mul) requires it and hence doesn't allow you to state the noncommutative version. But let's not worry about that now.